Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/CustomerReferral.hs

```diff
@@ -41,3 +63,93 @@ postPersonApplyReferral (mbPersonId, _) req = do
+      verifyVpaCall = TPayment.verifyVpa person.merchantId person.merchantOperatingCityId Nothing TPayment.Normal
+  resp <- try @_ @SomeException $ DP.verifyVPAService commonMerchantId commonPersonId verifyVPAReq verifyVpaCall
+  case resp of
+    Left e -> throwError $ InvalidRequest $ "VPA Verification Failed: " <> show e
+    Right response -> do
+      logDebug $ "verify vpa resp : " <> show response
+      pure $ VpaResp {vpa = response.vpa, isValid = response.status == "VALID"}
+
+getReferralPayoutHistory :: (Maybe (Id Person.Person), Id Merchant.Merchant) -> Flow PayoutHistory
+getReferralPayoutHistory (mbPersonId, _mbMerchantId) = do
+  personId <- mbPersonId & fromMaybeM (PersonNotFound "No person found")
+  payoutOrders <- QPayoutOrder.findAllByCustomerId personId.getId
+  let history = map getPayoutOrderDetails payoutOrders
+  pure $ PayoutHistory {..}
+  where
+    getPayoutOrderDetails payoutOrder =
+      PayoutItem
+        { amount = payoutOrder.amount.amount,
+          vpa = payoutOrder.vpa,
+          payoutAt = payoutOrder.createdAt,
+          payoutStatus = DPayout.castOrderStatus payoutOrder.status,
+          orderId = payoutOrder.orderId
+        }
+
+postPayoutVpaUpsert :: (Maybe (Id Person.Person), Id Merchant.Merchant) -> UpdatePayoutVpaReq -> Flow APISuccess
+postPayoutVpaUpsert (mbPersonId, _mbMerchantId) req = do
+  personId <- mbPersonId & fromMaybeM (PersonNotFound "No person found")
+  person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  QPerson.updatePayoutVpa (Just req.vpa) personId
+  fork ("processing backlog payout for customer while vpa updation" <> personId.getId) $
+    processBacklogReferralPayout personId req.vpa person.merchantOperatingCityId
+  pure Success
+
+processBacklogReferralPayout ::
+  ( CacheFlow m r,
+    EsqDBFlow m r,
+    MonadFlow m,
+    EncFlow m r
+  ) =>
+  Id Person.Person ->
+  Text ->
+  Id MerchantOpCity.MerchantOperatingCity ->
+  m ()
+processBacklogReferralPayout personId vpa merchantOpCityId = do
+  person <- QPerson.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  mbPayoutConfig <- CPC.findByCityIdAndVehicleCategoryAndIsPayoutEnabled person.merchantOperatingCityId True VehicleCategory.AUTO_CATEGORY
+  personStats <- PStats.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+  let toPayReferredByReward = personStats.referredByEarnings > 0 && isNothing personStats.referredByEarningsPayoutStatus
+      toPayBacklogAmount = personStats.backlogPayoutAmount > 0 && isNothing personStats.backlogPayoutStatus
+  when (toPayReferredByReward || toPayBacklogAmount) $ do
+    Redis.withWaitOnLockRedisWithExpiry (Common.payoutProcessingLockKey personId.getId) 3 3 $ do
+      let amount = (bool 0 personStats.backlogPayoutAmount toPayBacklogAmount) + (bool 0 personStats.referredByEarnings toPayReferredByReward)
+      case (toPayReferredByReward, toPayBacklogAmount) of
```

Comment:
[optional] We can remove this case match as it is similar to what we are doing in getEntityName
Do getEntityName above and based on entity we can do update
Won't have to think about booleans then